controller/src/main/java/io/pravega/controller/store/stream/StreamMetadataStore.java

```diff
@@ -1276,4 +1276,40 @@
     CompletableFuture<Integer> getSegmentSealedEpoch(final String scope, final String streamName, final long segmentId,
                                                      final OperationContext context, final Executor executor);
 
+    /**
+     * Compares two Stream cuts and returns true if streamcut1 is strictly ahead of streamcut2 else returns false. 
+     * This method will return false for both strictly less than and overlapping streamcuts.
+     * A streamcut is considered greater than if for all key ranges the segment/offset in one streamcut is ahead of 
+     * second streamcut. 
+     *
+     * @param scope      stream scope.
+     * @param streamName stream name.
+     * @param streamCut1 Stream cut to check
+     * @param streamCut2 Streamcut to check against
+     * @param context    operation context.
+     * @param executor   callers executor.
+     *                   
+     * @return A completable future which when completed will hold a boolean which will indicate if streamcut1 is strictly
+     * ahead of streamcut2. 
+     */
+    CompletableFuture<Boolean> streamCutStrictlyGreaterThan(final String scope, final String streamName,
+                                                            Map<Long, Long> streamCut1, Map<Long, Long> streamCut2,
```

Comment:
What is meant by "strictly" here? Would be good to explain that in the javadoc

Reply:
isnt this already explained in the javadoc here. 
```
     * Compares two Stream cuts and returns true if streamcut1 is strictly ahead of streamcut2 else returns false. 
     * This method will return false for both strictly less than and overlapping streamcuts.
     * A streamcut is considered greater than if for all key ranges the segment/offset in one streamcut is ahead of 
     * second streamcut. 
```